src/main/java/build/buildfarm/instance/memory/MemoryInstance.java

```diff
@@ -14,11 +14,17 @@
 
 package build.buildfarm.instance.memory;
```

Comment:
Yes, this does change the lines a bit. I fought with this logically and came up with the following:

The memory instance is responsible for retaining all ephemeral state in memory for the purpose of execution. It is using well known interfaces for access to the ContentAddressableStorage and ActionCache, which were quite literally set apart in the remote_execution service definition. It is incapable of continuing a watch for a currently executing Operation through termination, and workers which attempt to report back via the OperationQueueService  through a termination will be met with failure. Since he's not assuming responsibility for the CAS or AC that he is being passed, and not dropping those other memory-based limitations, it's not a convention break.